src/doors/TrainDoors.cpp

```diff
@@ -215,12 +214,12 @@ namespace godot {
         state["doors_right_remote_open"] = right_door.remote_open;
         state["doors_right_position"] = right_door.position;
         state["doors_right_position_normalized"] = right_door.position / max_shift;
-        state["doors_right_operating"] = right_door.is_opening | right_door.is_closing;
+        state["doors_right_operating"] = static_cast<int>(right_door.is_opening) | static_cast<int>(right_door.is_closing);
```

Comment:
changes bool to int. should be bool.

Reply:
I took more brief look into that

```cpp
right_door.is_opening | right_door.is_closing;
```

`|` Implies conversion `bool -> init` as `|` is not an `OR` operator. `||` is, so I presume `|| should be there`

Thanks for pointing this out